### PR TITLE
rpcsrv: carefully store Oracle service instance

### PR DESCRIPTION
**Pull request by AnnaShaleva**
### Problem

Invalid `submitoracleresponse` RPC handler behaviour in case of disabled Oracle service: no `neorpc.NewRPCError("Oracle is not enabled", "")` error is returned. Thanks to @tatiana-nspcc for the report.

### Solution

Check for typed nil value while setting Oracle service. I know, that we don't want to import the whole `oracle` package from the `rpcsrv`, but the alternatives I see looks worser to me:
  1. Check that Oracle service is a proper nil value each time when `rpcsrv.New(...)` is used. It's not very convinient; there are three places in the code and a couple more in tests.
  2. Use reflection inside `rpcsrv` to ensure that Oracle service is really nil. We'd better avoid reflection untill there are no other choices.

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
The test for the case of disabled Oracle service is added in the #3063.

**Comment by roman-khimov** [comment]
This makes `OracleHandler` interface completely useless. I'd just have `pythia rpcsrv.OracleHandler` in the code calling `rpcsrv.New` (just three real cases and one is always a proper `nil`) and initialize it accordingly (`mkOracle` is almost ready to return `rpcsrv.OracleHandler` as well, maybe an additional interface would solve it).

File: pkg/services/rpcsrv/server.go [hunk 1]
```diff
@@ -445,7 +446,9 @@ func (s *Server) Shutdown() {
 
 // SetOracleHandler allows to update oracle handler used by the Server.
 func (s *Server) SetOracleHandler(orc OracleHandler) {
-	s.oracle.Store(&orc)
+	if orc != nil {
```
Comment (roman-khimov) [inline pkg/services/rpcsrv/server.go:449]: It can be `nil`, suppose you have an oracle enabled, now you disable it.

File: cli/server/server.go [hunk 2]
```diff
@@ -359,7 +359,14 @@ func resetDB(ctx *cli.Context) error {
 	return nil
 }
 
-func mkOracle(config config.OracleConfiguration, magic netmode.Magic, chain *core.Blockchain, serv *network.Server, log *zap.Logger) (*oracle.Oracle, error) {
+// OracleService is an interface representing Oracle service with network.Service
+// capabilities and ability to submit oracle responses.
+type OracleService interface {
```
Comment (roman-khimov) [inline cli/server/server.go:364]: Likely doesn't need to be exported.

File: pkg/services/rpcsrv/server.go [hunk 3]
```diff
@@ -2489,7 +2491,7 @@ func (s *Server) submitOracleResponse(ps params.Params) (any, *neorpc.Error) {
 	if !pub.Verify(msgSig, hash.Sha256(data).BytesBE()) {
 		return nil, neorpc.NewRPCError("Invalid request signature", "")
 	}
-	(*oracle).AddResponse(pub, uint64(reqID), txSig)
+	(oracle).AddResponse(pub, uint64(reqID), txSig)
```
Comment (roman-khimov) [inline pkg/services/rpcsrv/server.go:2494]: `oracle.AddResponse()`?

**Review by roman-khimov — APPROVED** [review]

File: cli/server/server.go (hunk 2 above)
Comment (AnnaShaleva) [reply cli/server/server.go]: Yes, fixed.

File: pkg/services/rpcsrv/server.go (hunk 3 above)
Comment (AnnaShaleva) [reply pkg/services/rpcsrv/server.go]: Sure.

File: pkg/services/rpcsrv/server.go [hunk 4]
```diff
@@ -2461,10 +2462,11 @@ func getRelayResult(err error, hash util.Uint256) (any, *neorpc.Error) {
 }
 
 func (s *Server) submitOracleResponse(ps params.Params) (any, *neorpc.Error) {
-	oracle := s.oracle.Load().(*OracleHandler)
-	if oracle == nil || *oracle == nil {
+	oraclePtr := s.oracle.Load()
+	if oraclePtr == nil {
```
Comment (tatiana-nspcc) [inline pkg/services/rpcsrv/server.go:2466]: Cool! Thank you for fixing

**Review by tatiana-nspcc — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**